### PR TITLE
Mushmen fixes

### LINKED ISSUE #22141: Moshroom Language
(WEB REPORT BY: avadanatinde REMOTE: 206.221.180.138:7777)
# Revision
d77d0d64c9367122d57ec493349bb10a00987240

# Description
Moshrooms cannot understand Vox Pidgin as their adult form. This is a race that is entirely integrated with the vox to the point where the vox traders have space suits for their moshroom friends. While they are a race that is unable to speak (they use telepathy and understand galactic common), they should be able to understand vox-pidgin at the very least with how integrated with the vox they are.
# Steps to Reproduce
Become Mushroom person
Listen to Vox squaking and not be able to understand it
# What you Expected
To be able to understand the language of the vox.
# What Actually Happened
You are unable to understand vox-pidgin.

### LINKED ISSUE #22652: Mushmen can gain regular sight while wearing an active mushman helmet by removing and wearing it again
#### Description of issue
Mushmen get fucked in the light and have a helmet that basically works as a glorified welding mask to protect them, complete with obnoxious screen obscuring effect.
This effect can be removed by merely taking off the helmet, waiting for it to go away and placing it back on again.

#### Difference between expected and actual behavior
**Expected:**
- Remove helmet, screen obstruction goes away
- Replace helmet, screen obstruction returns

**Actual**
- Remove helmet, screen obstruction goes away
- Replace helmet, screen obstruction remains gone

#### Steps to reproduce

- Be mushman
- Put on mushman helmet and activate it's screen
- Take off the helmet and wait until the overlay goes away
- Put it back on

#### Server revision

716b4b042f245ae472a465c7dbf2b8cffacb9598

### LINKED ISSUE #22654: Moshroom monkey tails
(WEB REPORT BY: avadanatinde REMOTE: 206.221.180.138:7777)
# Revision
716b4b042f245ae472a465c7dbf2b8cffacb9598

# Description
The walking mushroom (602) waves his tail.
these are mushrooms not monkeys
# Steps to Reproduce
plant a moshroom seed and harvest it
observe the walking mushroom shake it's tail occasionally
# What you Expected
them to not have tails
# What Actually Happened
apparently they have tails

### PR DESCRIPTION
**Pull request by ProfanedBane**
Fixes:
- sporemind using your visible name rather than your real_name
- babby mushroom wagging a non-existent tail, #22654 
- the overlay for the mushman helmet being nonfunctional at times, #22652 
- mushmen do not know voxspeak, #22141 
- mushman spacesuit was named mushman helmet

closes #22654 closes #22652 closes #22141

:cl: 
 - tweak: mushrom sporemind now shows real names rather than visible names
 - bugfix: mushroms no longer waggle their non-existent tails
 - bugfix: fixed mushman helmet overlays breaking
 - rscadd: mushmen can understand Vox Pidgin
 - spellcheck: mushman spacesuit was named mushman helmet

<!--
Pull requests must be atomic.  Change one set of related things at a time.  Bundling sucks for everyone.
This means, primarily, that you shouldn't fix bugs and add content in the same PR. When we mean 'bundling', we mean making one PR for multiple, unrelated changes.

Test your changes. PRs that do not compile will not be accepted.
Testing your changes locally is incredibly important. If you break the serb we will be very upset with you.

Large changes require discussion.  If you're doing a large, game-changing modification, or a new layout for something, discussion with the community is required as of 26/6/2014.  Map and sprite changes require pictures of before and after.  MAINTAINERS ARE NOT IMMUNE TO THIS.  GET YOUR ASS IN IRC.

Merging your own PRs is considered bad practice, as it generally means you bypass peer review, which is a core part of how we develop.

It is also suggested that you hop into irc.rizon.net #vgstation or the coding discord (you can find an invite link on the irc or the thread) to discuss your changes, or if you need help.

== SELF LABELLING PRs ==
You can now self-label your PR! The syntax is simple. Just put [<labeltag>] anywhere in your PR,
where <labeltag> is to be replaced by one of the following (allowing with the resultant tag)
just ctrl+f the list or something. I know it's long.

administration = "Logging / Administration"
away = "Mapping (Away Mission :earth_africa:)"
bagel = "Mapping (Bagel :o:)"
box = "Mapping (Box :baby:)"
bugfix = "Bug / Fix"
bus = "Mapping (Bus :bus:)"
byond = "T-Thanks BYOND"
consistency = "Consistency Issue"
controversial = "Controversial"
deff = "Mapping (Deff :wastebasket:)"
discussion = "Discussion"
dnm = "✋ Do Not Merge ✋"
easy = "Easy Fix"
exploitable = "Exploitable"
featureloss = "Feature Loss"
featurerequest = "Feature Request"
first = "good first issue"
formatting = "Grammar / Formatting"
gamemode = "Gameplay / Gamemode"
gameplay = "Gameplay / Gamemode"
general = "Mapping (General :world_map:)"
goonchat = "Goonchat"
grammar = "Grammar / Formatting"
help = "help wanted"
hotfix = "Hotfix"
libvg = "libvg"
logging = "Logging / Administration"
meta = "Mapping (Meta :no_mobile_phones:)"
needspritework = "Needs Spritework"
oversight = "Oversight"
packed = "Mapping (Packed :package:)"
parallax = "Parallax"
qol = "❤️ Quality of Life ❤️"
roid = "Mapping (Roidstation :pick:)"
role = "Role Datums"
roleissue = "Role Datums Issue"
runtime = "Runtime Log"
sanity = "Sanity / Ghosthands"
snowmap = "Mapping (Snowmap ❄)"
sound = "Sound"
sprites = "Sprites"
spriteworkdone = "Spritework Done Needs Coder"
system = "System"
taxi = "Mapping (Taxi :taxi:)"
tested = "100%  tested"
tweak = "Tweak"
ui = "UI"
vault = "Mapping (Vault :question:)"
vote = "⛔ Requires Server Vote ⛔"
wip = "WiP"

[bugfix]
[qol]

:cl:
 * bugfix: Fix exploit with the mushman helmet overlay, mushkeys wagging their nonexistent tail, the mushman space suit being named incorrectly
 * tweak: Mushman sporemind uses real_name instead of your visible name, mushmen now understand voxspeak

### CODE REVIEW

**Review by ShiftyRail — APPROVED** [review]
Thanks!

Don't forget the :cl: as well as "closes #xxxxx"

**Comment by ProfanedBane** [comment]
Edited in the closes stuff, changelog is at the bottom since it says to put it there.

**Comment by SonixApache** [comment]
Nice fixes/changes. 👌 

Your syntax is off, put the following in your OP if you want to quick-fix it:
```
closes #22654 closes #22652 closes #22141
:CL:
 - tweak: mushrom sporemind now shows real names rather than visible names
 - bugfix: mushroms no longer waggle their non-existent tails
 - bugfix: fixed mushman helmet overlays breaking
 - rscadd: mushmen can understand Vox Pidgin
 - spellcheck: mushman spacesuit was named mushman helmet
```

File: code/modules/clothing/spacesuits/alien.dm [hunk 1]
```diff
@@ -222,22 +222,21 @@
...
+	var/up = 0
 
 /obj/item/clothing/head/helmet/space/vox/civ/mushmen/attack_self(var/mob/user)
 	toggle(user)
 
 /obj/item/clothing/head/helmet/space/vox/civ/mushmen/proc/toggle(var/mob/user)
 	if(!user.incapacitated())
-		if(!src.up)
+		if(src.up)
 			to_chat(user, "<span class='notice'>You use \the [src]'s visor to protect your face from incomming light.</span>")
```
Comment (MadmanMartian) [inline code/modules/clothing/spacesuits/alien.dm]: should be incoming, not incomming

File: code/modules/clothing/spacesuits/alien.dm [hunk 2]
```diff
@@ -222,22 +222,21 @@
...
-			user.overlay_fullscreen("tint", /obj/abstract/screen/fullscreen/impaired, 2)
 		else
 			to_chat(user, "<span class='notice'>You disengage \the [src] light protection visor.</span>")
```
Comment (MadmanMartian) [inline code/modules/clothing/spacesuits/alien.dm]: \the [src]'s

**Comment by MadmanMartian** [comment]
good changes 👍 

The mushroom helmet's ability is still somewhat misleading, but it'd require a bit of a rewrite of how REQUIRE_DARK and REQUIRE_LIGHT work, to get body coverage.

**Comment by ProfanedBane** [comment]
Added in the requested typo fixes. And ye there's a couple other things I'm wanting to do at some point but they're out of the scope of simple bug fixes.

**Comment by ShiftyRail** [comment]
So this is good to go ?

**Comment by MadmanMartian** [comment]
Honestly, the helmet needs to have its snowflake welding stuff removed and just have /datum/action/item_action/toggle_helmet_mask added to its action_types, as this is just piling onto the spaghetti code at this point.

**Comment by ProfanedBane** [comment]
If that's a worthwhile change may aswell leave this unmerged till I get round to it, 2late today but I'll have a look later. Not had a look but I assume "toggle_helmet_mask" handles the overlay shit automagically?

File: code/modules/mob/living/carbon/human/life/handle_regular_hud_updates.dm [hunk 3]
```diff
@@ -306,17 +306,19 @@
 		var/masked = 0
 
 		if(head)
-			if(istype(head, /obj/item/clothing/head/welding) || istype(head, /obj/item/clothing/head/helmet/space/unathi) || (/datum/action/item_action/toggle_helmet_mask in head.actions_types))
+			if(istype(head, /obj/item/clothing/head/welding) || istype(head, /obj/item/clothing/head/helmet/space/vox/civ/mushmen) || istype(head, /obj/item/clothing/head/helmet/space/unathi) || (/datum/action/item_action/toggle_helmet_mask in head.actions_types))
```
Comment (MadmanMartian) [inline code/modules/mob/living/carbon/human/life/handle_regular_hud_updates.dm:309]: Yeah, it makes this check true, so the overlay stuff will work properly.

**Comment by ProfanedBane** [comment]
Remove speedmergers >:(

**Comment by ProfanedBane** [comment]
@MadmanMartian Implementing that'll need equal amounts of spaghetti without making a generic "toggle helmet" action that takes fluff info etc from an object. And t b h I can't be bother to do that for this pr. Is it a must have or can we leave it be and make an issue so we remember the shitcode is there?

**Comment by MadmanMartian** [comment]
I'd say look into it, as it's removing fluff, not adding it. I'm not the final say in this though, so do whatever.

**Comment by Kurfursten** [comment]
I don't expect you to do that in the context of this PR, but Martian is right.
